src/main/php/PHPMD/Cache/Model/ResultCacheState.php

```diff
@@ -60,16 +60,16 @@ public function setViolations($filePath, array $violations): void
     public function addRuleViolation($filePath, RuleViolation $violation): void
     {
         $this->state['files'][$filePath]['violations'][] = [
-            'rule'          => get_class($violation->getRule()),
+            'rule' => get_class($violation->getRule()),
```

Comment:
This should work right?
```suggestion
            'rule' => ($violation->getRule())::class,
```

Reply:
Yes, it does. But out of scope.

Adjusted in https://github.com/phpmd/phpmd/pull/1133